base/math.jl

```diff
@@ -564,10 +564,10 @@ Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 function hypot(x::Number...)
     length(x) == 0 && error("invalid hypot() call without arguments")
     # speculatively try fast naive approach
-    s = sum(abs2 ∘ _floatunit, x)
+    s = sum(abs2 ∘ float, x)
     isnan(s) && return any(isinf, x) ? oftype(sqrt(s), Inf) : sqrt(s) # IEEE 754
-    fs = float(s)
-    if isinf(s) || fs ≤ _floatmin(fs) * (length(x) - 1)
+    sf = AbstractFloat(s)
+    if isinf(s) || sf ≤ _floatmin(sf) * (length(x) - 1)
```

Comment:
`AbstractFloat(s)` won't necessarily exist.  If you want to fix the units in the comparison, you can do `s ≤ oneunit(s) * _floatmin(sf) * (length(x) - 1)`

Reply:
Done.